src/Angor/Shared/Services/IndexerService.cs

```diff
@@ -225,5 +237,50 @@ public async Task<string> GetTransactionHexByIdAsync(string transactionId)
 
             return info;
         }
+
+        public async Task<string> GetGenesisBlockHashAsync(string indexerUrl)
+        {
+            try
+            {
+                var response = await _httpClient.GetAsync(indexerUrl);
+                response.EnsureSuccessStatusCode();
+                
+                var responseContent = await response.Content.ReadAsStringAsync();
+                return responseContent.Trim();
+
+
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error fetching genesis block hash: {ex.Message}");
+                return string.Empty;
+            }
+        }
+
+        public bool ValidateGenesisBlockHash(string fetchedHash, string expectedHash)
+        {
+            // Compare the first 64 characters (32 bytes in hex) of the fetched hash to the expected hash
+            return fetchedHash.StartsWith(expectedHash, StringComparison.OrdinalIgnoreCase);
+        }
+
+
+        public async Task<bool> CheckIndexerStatusAsync(string indexerUrl)
```

Comment:
I think we already have a way to check the indexer status look in the settings page for how we do that.